### PR TITLE
Add unit testing

### PR DESCRIPTION
**Pull request by aaronj1335**
This uses Jest for testing. You can test either by running jest
directly:

    npx jest

Or via `npm` (for CI integration):

    npm test

I tested this by updating my script and doing "Process now" from my
Gmail Automata sheet.

### CODE REVIEW

**Comment by aaronj1335** [comment]
Also, if you're curious, even though https://github.com/google/clasp#features says that modules are not supported, it apparently still works. I believe it is because of the TypeScript `esModuleInterop` setting.

File: .claspignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .git/**
 .idea/**
 build/**
-node_modules/**
\ No newline at end of file
+node_modules/**
+__tests__/**
```
Comment (ranmocy) [inline .claspignore:5]: I think it would be easier to find the test file by putting them along with the production file.
And normally those test files are named as `NAME.test.ts`. And you could filter them like `*.test.*` here.

**Comment by aaronj1335** [comment]
I saw you added a travis config, so I updated it to the version of node that works with this test.

How do you feel about using GitHub actions instead? I adore Travis. Don't get me wrong. I once gave a talk at a jQuery conference about building/deploying websites with it like 20 years ago https://github.com/aaronj1335/shipit . 

My argument for GitHub actions though would be it's one less click to see results from the pull request page: https://github.com/aaronj1335/gmail-automata/pull/4/checks?check_run_id=768652448 . I kinda like the left bar as well, since, if we add a "deploy" action (i.e. on push to master), we can separate those into different tabs where it's clear that one is testing and one is deploying.

**Comment by aaronj1335** [comment]
I guess another argument for GitHub is that forks inherit the tests by default, instead of also having to go to Travis and flip the switch for the repo.

File: .claspignore (hunk 1 above)
Comment (aaronj1335) [reply .claspignore:5]: Good call much nicer.

**Comment by ranmocy** [comment]
Github Actions seem interesting. I just set it up and let's give it a try.

**Comment by aaronj1335** [comment]
Ok I tweaked the file w/ some different names because that seems to look nicer in the actions UI to me, but LMK if you want something different.

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
-
-name: Node.js CI
+name: Checkout, install deps & test
```
Comment (ranmocy) [inline .github/workflows/node.js.yml:3]: Since we are using the squash merging strategy in this project, could you split this out into a separate PR so that each one could focus on doing one thing?

File: .travis.yml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 language: node_js
 node_js:
-  - 8
+  - 12
```
Comment (ranmocy) [inline .travis.yml:3]: Since we have no way to simulate true AppScript runtime, I think maybe we should just use "node" for the latest version?

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (aaronj1335) [reply .github/workflows/node.js.yml]: Done, can mail that one out later.

File: .travis.yml (hunk 3 above)
Comment (aaronj1335) [reply .travis.yml:3]: `8` doesn't work with the current version of Jest. Based on the travis docs, I think we actually have to list a number here, but am I mistaken?

Comment (ranmocy) [reply .travis.yml:3]: Instead of `- 12`, if you write `- node` it will always use the latest stable version of nodejs. So that we don't need to keep bumping the version.
https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions

Comment (aaronj1335) [reply .travis.yml]: 🤦 now i see!

**Comment by ranmocy** [comment]
Thanks for your contribution! Fix #22
